test/integration/integration-test-bootstrap.sh

```diff
@@ -269,56 +282,210 @@ function installProwCIDependencies {
 }
 
 function profileStage {
+    echo -e "\n========================================================================"
+    echo "= Int Testing Bootstrap Profile - Stage Cluster"
+    echo "========================================================================"
+    
+    echo "Configuring local build environment"
+    export FORCE_DEV_MODE=cluster
+    sourceEnvrcConfig
+    sanityCheck
+    
+    echo "Configuring depoloyment cluster"
     clusterUserName=$($OC whoami)
     ## OSD Staging cluster require cluster-admin roles for accessing & applying some manifests like CRDs etc.
     ## So, cluster-admin role is added to the user for the script's lifetime.
     ## The role is removed as the part of mandatory cleanup.
     $OC --as backplane-cluster-admin adm policy add-cluster-role-to-user cluster-admin $clusterUserName
-    export FORCE_DEV_MODE=cluster
-    sourceEnvrcConfig
-    sanityCheck
     cleanupPre
-    trap cleanupPost EXIT
-    $OC adm new-project "$NAMESPACE" || true
+    $OC adm new-project "$NAMESPACE" 2>/dev/null || true
     make prow-ci-predeploy
     make validate-deployment
+
+    echo "Building and deploying operator image"
     buildOperatorImage
     verifyBuildSuccess
     deployOperator
     waitForDeployment
 }
 
+function execWithTimeout {
+    local testScript=$1
+    local phase=$2
+    local timeout=$INT_TEST_PHASE_TIMEOUT #from test_envs
+
+    echo "========================================================================"
+    echo "= Test: $testScript"
+    echo "= Phase: $phase"
+    echo "========================================================================"
+
+    while true; do
+        timeout=$((timeout-1))
+        echo "$testScript $phase (timeout in: $timeout seconds)"
+
+        /bin/bash $testScript $phase
+        local exitCode=$?
+
+        if [ $exitCode -eq 0 ] || [ $exitCode -ne $EXIT_RETRY ]; then
+            return $exitCode
+        elif [ $timeout -le 0 ]; then
+            echo "ERROR - $testScript $phase timed out"
+            return $EXIT_TIMEOUT
+        else
+            echo "RETRYING $testScript $phase in 1 second"
+            sleep 1
+        fi
+    done
+}
+
+function explainExitCode {
+    local script=$1
+    local phase=$2
+    local exitCode=$3
+    message="UNKNOWN"
+
+    if [ -z $exitCode ]; then
+        message="No test results found"
+    elif [ $exitCode -ne 0 ]; then
+        message=$(/bin/bash $script explain $phase exitCode)
+        if [ -z "$message" ]; then
+            # check the general exit code messages
+            message=${GENERAL_EXIT_CODE_MESSAGES[$exitCode]}
+            if [ -z "$message" ]; then
+                message="UNKNOWN"
+            fi
+        fi
+    else
+        message="PASS"
+    fi
+    echo $message
+}
+
+
+function runTest {
+    local testScript=$1
+    overall="PASS"
+
+    
+    execWithTimeout $testScript "setup"
+    setupExitCode=$?
+    setupExitMessage=$(explainExitCode $testScript "setup" $setupExitCode)
```

Comment:
we can remove passing the "phase" to all of the `explainExitCode` calls here too.